### PR TITLE
Add pyenv install script and clean up env setup docs

### PR DESCRIPTION
**Pull request by vladkluev**
# Problem
pyenv does not install python3.8.3 properly on big sur, making the dev env hard to set up

# Solution
* port the [script](https://github.com/getsentry/sentry/blob/master/scripts/pyenv_setup.sh) from getsentry/sentry into this repo (with its dependencies)
* update the docs to remove virtualenvwrapper requirement from the environment setup

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,5 +1,9 @@
 .PHONY: develop setup-git test install-python-dependencies
 
+pyenv-setup:
+	@./scripts/pyenv_setup.sh
+	pip install --upgrage pip==21.1.3
```
Comment (rahul-kumar-saini) [inline Makefile:5]:
```suggestion
	pip install --upgrade pip==21.1.3
```

**Review by rahul-kumar-saini — CHANGES_REQUESTED** [review]

File: scripts/lib.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# Module containing code shared across various shell scripts
+# Execute functions from this module via the script do.sh
+
+# Check if a command is available
+require() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+configure-sentry-cli() {
```
Comment (evanh) [inline scripts/lib.sh:10]: Is this function necessary? I know that it's called by the pyenv script but does it need to be in Snuba?

Comment (vladkluev) [reply scripts/lib.sh:10]: The pyenv setup script requires it

https://github.com/getsentry/snuba/pull/2012/files#diff-412b37f70434d5f5ec589787dda24887d69ec454a6d53045a9459edf0478089bR112

I could try and see if it'll work without it but I assumed that it was in there for a reason (so I kept it)

Comment (evanh) [reply scripts/lib.sh:10]: Yeah I saw that it's in the pyenv script, I'm just not sure why it's necessary. `sentry-cli` isn't called anywhere in the script.

Comment (fpacifici) [reply scripts/lib.sh:10]: I think this is used here to send errors to sentry during the install script
https://docs.sentry.io/product/cli/send-event/#bash-hook
Adding a dependency on a sentry tool on the snuba code base for that (considering we only set up the python env) for that seem a bit overkill but I won't block it for that reason

Comment (evanh) [reply scripts/lib.sh:10]: Oh I see. Yeah OK well that makes sense.

**Review by evanh — APPROVED** [review]

**Review by rahul-kumar-saini — APPROVED** [review]

**Comment by armenzg** [comment]
Hey, I'm glad the code could be re-used, however, please try to reach me or Billy to make your life simpler.

Moving from Python 3.8.3 to 3.8.10 would have been simpler since that Python version works via pyenv on Big Sur.

Also, a lot of the pyenv configuration was already executed on first run for new Sentry engineers as part of the bootstrapping code.
https://github.com/getsentry/bootstrap-sentry/blob/8c2cf34115b541034579e93b1104be02ed34a97d/bootstrap.sh#L401-L413

Feel free to reach out or tag our team. We'll be happy to assist.

**Pull request merged**